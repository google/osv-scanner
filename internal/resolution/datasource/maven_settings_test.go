package datasource_test

import (
	"testing"

	"github.com/google/osv-scanner/internal/resolution/datasource"
	"github.com/google/osv-scanner/internal/testutility"
)

func TestParseMavenSettings(t *testing.T) {
	t.Setenv("MAVEN_SETTINGS_TEST_USR", "UsErNaMe")
	t.Setenv("MAVEN_SETTINGS_TEST_PWD", "P455W0RD")
	t.Setenv("MAVEN_SETTINGS_TEST_SID", "my-cool-server")
	t.Setenv("MAVEN_SETTINGS_TEST_NIL", "")
	s := datasource.ParseMavenSettings("./fixtures/maven_settings/settings.xml")
	testutility.NewSnapshot().MatchJSON(t, s)
}

func TestMakeMavenAuth(t *testing.T) {
	t.Parallel()
	globalSettings := datasource.MavenSettingsXML{
		Servers: []datasource.MavenSettingsXMLServer{
			{
				ID:       "global",
				Username: "global-user",
				Password: "global-pass",
			},
			{
				ID:       "overwrite1",
				Username: "original-user",
				Password: "original-pass",
			},
			{
				ID:       "overwrite2",
				Username: "user-to-be-deleted",
				// no password
			},
		},
	}
	userSettings := datasource.MavenSettingsXML{
		Servers: []datasource.MavenSettingsXMLServer{
			{
				ID:       "user",
				Username: "user",
				Password: "pass",
			},
			{
				ID:       "overwrite1",
				Username: "new-user",
				Password: "new-pass",
			},
			{
				ID: "overwrite2",
				// no username
				Password: "lone-password",
			},
		},
	}

	mAuth := datasource.MakeMavenAuth(globalSettings, userSettings)
	testutility.NewSnapshot().MatchJSON(t, mAuth)
}
