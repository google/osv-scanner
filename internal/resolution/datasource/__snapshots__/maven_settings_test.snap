
[TestMakeMavenAuth - 1]
{
  "global": {
    "SupportedMethods": [
      2,
      0
    ],
    "AlwaysAuth": false,
    "Username": "global-user",
    "Password": "global-pass",
    "BasicAuth": "",
    "BearerToken": ""
  },
  "overwrite1": {
    "SupportedMethods": [
      2,
      0
    ],
    "AlwaysAuth": false,
    "Username": "new-user",
    "Password": "new-pass",
    "BasicAuth": "",
    "BearerToken": ""
  },
  "overwrite2": {
    "SupportedMethods": [
      2,
      0
    ],
    "AlwaysAuth": false,
    "Username": "",
    "Password": "lone-password",
    "BasicAuth": "",
    "BearerToken": ""
  },
  "user": {
    "SupportedMethods": [
      2,
      0
    ],
    "AlwaysAuth": false,
    "Username": "user",
    "Password": "pass",
    "BasicAuth": "",
    "BearerToken": ""
  }
}
---

[TestParseMavenSettings - 1]
{
  "Servers": [
    {
      "ID": "server1",
      "Username": "user",
      "Password": "pass"
    },
    {
      "ID": "server2",
      "Username": "UsErNaMe",
      "Password": "~~P455W0RD~~"
    },
    {
      "ID": "my-cool-server",
      "Username": "${env.maven_settings_test_usr}-",
      "Password": "${env.MAVEN_SETTINGS_TEST_BAD}"
    }
  ]
}
---
